### PR TITLE
Adjust inflation payout based on era duration

### LINKED ISSUE #393: inflation curve
use the inflation curve to accurately model target stake %, APY, and handle shorter era durations

### PR DESCRIPTION
**Pull request by jordy25519**
Related to missed payouts as part of #632 

Changes:
- Adjust inflation payouts based on era duration
- Forced era payouts are always scheduled to be paid
- Scheduling uses `on_idle` instead of predetermined block numbers

Other:
- spiked into using inflation curve but values aren't quite right. closes #393 as a result

TODO:
- [x] fix integration tests
- [ ] update rewards section in wiki

### CODE REVIEW

**Review by JasonTulp — APPROVED** [review]
lgtm!

File: runtime/tests/staking_reward.rs [hunk 1]
```diff
@@ -285,96 +295,36 @@ fn era_transaction_fees_collected() {
...
 
 			// At the start of the next era, transaction rewards should be cleared
-			assert!(Rewards::calculate_total_reward().transaction_fees.is_zero());
+			assert!(Rewards::calculate_total_reward(era_duration_ms)
+				.transaction_fees
+				.is_zero());
 		});
 }
 
 #[test]
```
Comment (jordy25519) [inline runtime/tests/staking_reward.rs:328]: accruing doesn't happen anymore, payouts happen as normal

File: runtime/src/impls.rs [hunk 2]
```diff
@@ -608,6 +609,7 @@ mod tests {
 	}
 
 	#[test]
+	#[ignore]
```
Comment (jordy25519) [inline runtime/src/impls.rs:612]: this test isn't really useful and adds ~2minutes to CI

**Review by JasonTulp — APPROVED** [review]
Looks good!

**Pull request merged**